lib/utils.js

```diff
@@ -68,45 +43,34 @@ const getDataStats = (data, checksum, dedupHash) => ({
   size: Buffer.byteLength(data),
 });
 
-const compress = (file, minCompressSize, compression, cb) => {
-  fs.stat(file, (err, stats) => {
-    if (err || stats.size <= minCompressSize) {
-      cb(err, false);
-      return;
-    }
-    const filec = file + 'z';
-    const compressor = compressors[compression];
-    if (!compressor) {
-      throw new Error(`Unknown compression type ${compression} specified`);
-    }
-    compressor
-      .compressFile(file, filec)
-      .then(() =>
-        fs.rename(filec, file, err => {
-          if (err) cb(err, false);
-          else cb(null, true);
-        })
-      )
-      .catch(err => cb(err, false));
-  });
+const compress = async (file, minCompressSize, compression) => {
+  const compressor = compressors[compression];
+  if (!compressor) {
+    throw new Error(`Unknown compression type ${compression} specified`);
+  }
+
+  const stats = await fs.stat(file);
+  if (stats.size <= minCompressSize) return false;
+
+  const filec = file + 'z';
+  await compressor.compressFile(file, filec);
+  await fs.rename(filec, file);
+  return true;
 };
 
-const uncompress = (file, opts, cb) => {
-  fs.access(file, err => {
-    if (err) {
-      cb(err);
-      return;
-    }
-    const compressor = compressors[opts.compression];
-    if (!compressor) {
-      throw new Error(`Unknown compression type ${opts.compression} specified`);
-    }
+const uncompress = async (file, opts) => {
+  const compressor = compressors[opts.compression];
+  if (!compressor) {
+    throw new Error(`Unknown compression type ${opts.compression} specified`);
+  }
+
+  return new Promise((res, rej) => {
     const buffers = [];
     new compressor.UncompressStream({ source: file })
-      .on('error', cb)
-      .on('finish', () => {
-        if (opts.encoding) cb(null, buffers.join());
-        else cb(null, Buffer.concat(buffers));
+      .once('error', rej)
+      .once('finish', () => {
+        if (opts.encoding) res(buffers.join(''));
+        else res(Buffer.concat(buffers));
       })
       .on('entry', (header, stream, next) => {
```

Comment:
I don't think that this will work with gzip stream, since it doesn't emit `'entry'` event:

> **Note: tar, tgz and zip have the same uncompressing streaming API as above: it's a writable stream, and entries will be emitted while uncompressing one after one another, while that of gzip is slightly different: gzip.UncompressStream is a transform stream, so no entry event will be emitted and you can just pipe to another stream**

Also, I think, you should just use the newly added [common.MemoryWritable](https://github.com/metarhia/common#class-memorywritable-extends-writable) here.